src/framework/pybind_casts.hpp

```diff
@@ -0,0 +1,70 @@
+/**
+ * This code is part of Qiskit.
+ *
+ * (C) Copyright IBM 2021.
+ *
+ * This code is licensed under the Apache License, Version 2.0. You may
+ * obtain a copy of this license in the LICENSE.txt file in the root directory
+ * of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Any modifications or derivative works of this code must retain this
+ * copyright notice, and modified files need to carry a notice indicating
+ * that they have been altered from the originals.
+ */
+
+#ifndef _aer_framework_pybind_casts_hpp_
+#define _aer_framework_pybind_casts_hpp_
+
+#include "../simulators/stabilizer/clifford.hpp"
+
+namespace py = pybind11;
+
+namespace pybind11 {
+namespace detail {
+template <typename T> struct type_caster<matrix<T>>{
+  using base = type_caster_base<matrix<T>>;
+public:
+  PYBIND11_TYPE_CASTER(matrix<T>, _("matrix_complex_t"));
```

Comment:
Should this be
```suggestion
  PYBIND11_TYPE_CASTER(matrix<T>, _("matrix_t"));
```
Since matrix doesnt need to be complex